【Tutorial 6.2 - API（Firebase）連携を学ぶ】 の実装

k-shimizu3 opened the pull request:
# Tutorial 6.2
### 課題
DockerとFirebaseを使用して、JSONファイルをインターネットに公開
不動産アプリの画面模写で使われているダミーデータをAPIの使用に書き換える

---

### 主な実装内容
前節（Tutorial 5）を参考にMVVMで実装を行いました。
freezed・retrofit・riverpodパッケージを使用しています。

-  lib / main.dart
   - ProviderScopeの追加

 - lib / residence / 
   - residence_client_state_notifier.dart
   - api / residence_api_client.dart
   - model / residence_metadata.dart
   - repository / residence_repository.dart
   - state / residence_client_state.dart
   - residence_screen.dart
     - ダミーデータからAPIの使用に変更
   - data_class / residence_metadata.dart (削除)
     - モデルを追加したため削除

 - pubspec.yaml
    - freezed・retrofit・riverpodパッケージ等の追加
    - intl の使用
       - 不動産価格にカンマを入れるのに使用

---
### 動作サンプル
不動産情報をダミーデータからAPIによる取得に変更した為、コードを増やすことなく表示データを増やすことが出来ました

https://user-images.githubusercontent.com/114896398/197474728-b1671a90-8151-40e3-9746-ef29001faf98.mov

---

### 課題（悩んだこと）
特にありません。

---

### 備考
いつもレビューありがとうございます。
レビューよろしくお願いします！

k-shimizu3 committed:
パッケージの追加・riverpodのProviderScope()の追加

k-shimizu3 committed:
不動産情報をダミーデータからAPIの取得に変更

MVVM・各パッケージ（freezed・retrofit・riverpod）を使用して実装を行いました

k-shimizu3 committed:
自動生成ファイルの追加

k-shimizu3 committed:
developブランチをマージ

先ほどGitHub上で feature/tutorial4ブランチを developブランチへマージしたので、その分の変更をこのブランチにも反映させました。

onishitaisei commented:
@kenji-axia 
LGTMです！！
ただ添付していただいている動画について、ボタンを押したときに一瞬画面が止まっているのでどこか重い処理を同期的に行っているのではないかと思いました！時間があったら別でissue切って原因を探ってみるのも勉強になって良いかなと思います！

k-shimizu3 committed:
先に進んだdevelopブランチをマージ

コンフリクト対策です。

Merge branch 'develop' into feature/tutorial6.2

# Conflicts:
#	pubspec.yaml
